app/src/main/java/com/example/bookwormadventuresdeluxe2/LoginActivity.java

```diff
@@ -89,4 +145,157 @@ public void onClick(View view)
+                                                {
+                                                    for (QueryDocumentSnapshot snapshot : queryDocumentSnapshots)
+                                                    {
+                                                        // add to UserCredentialAPI to be accessible throughout app
+                                                        UserCredentialAPI userCredentialAPI = UserCredentialAPI.getInstance();
+                                                        userCredentialAPI.setUsername(snapshot.getString("username"));
+                                                        userCredentialAPI.setUserId(snapshot.getString("userId"));
+                                                    }
+                                                    progressBar.setVisibility(View.INVISIBLE);
+                                                    // Go to ListActivity
+                                                    startActivity(new Intent(LoginActivity.this, MyBooksActivity.class));
+                                                }
+                                            }
+                                        });
+                            }
+                            else
+                            {
+                                String errorCode = "";
+
+                                /* Extract Firebase Error Code */
+                                try
+                                {
+                                    errorCode = ((FirebaseAuthException) task.getException()).getErrorCode();
+                                } catch (Exception e)
+                                {
+                                    String tooManyLogins = task.getException().getMessage();
+                                    Toast.makeText(LoginActivity.this, tooManyLogins, Toast.LENGTH_LONG).show();
+                                }
+
+                                /* Set EditText Error type from errorCode */
+                                switch (errorCode)
+                                {
+                                    case "ERROR_WRONG_PASSWORD":
+                                        EditTextValidator.wrongPassword(editTextPassword);
+                                        progressBar.setVisibility(View.INVISIBLE);
+                                        break;
+
+                                    case "ERROR_INVALID_EMAIL":
+                                        /* Merged with next case as both set error to editTextEmail*/
+                                    case "ERROR_USER_NOT_FOUND":
+                                        EditTextValidator.emailNotFound(editTextEmail);
+                                        progressBar.setVisibility(View.INVISIBLE);
+                                        break;
+
+                                    default:
+                                        break;
```

Comment:
Throw exception if invalid error code! We dont want this to silently fail

Reply:
added.